src/services/user.js

```diff
@@ -35,11 +37,19 @@ export default class UserService {
 
   static async deleteUser(id) {
     const isUserExist = await findUserById(id);
-    if (isUserExist) {
-      await deleteUser(id);
-    } else {
+    if (!isUserExist) {
       throw new NotFoundError(`User with id = ${id} cannot be deleted`);
     }
+
+    const transaction = await sequelize.transaction();
+    try {
+      await deleteUser(id, transaction);
+      await deleteUserFromGroups(id, transaction);
```

Comment:
By the way, if I'm not mistaken postgres can automatically delete all connected to user rows when user is deleted. Please look into `CASCADE` delete

Reply:
Yes, but we don't delete user from Users table. We just mark it `isDeleted: true`. So we should remove users from UserGroup table manually.